src/lazy.jl

```diff
@@ -627,14 +627,14 @@ isbigfloat(x::NumberResult) = x.tag == BIGFLOAT
             # if we overflowed, then let's try BigFloat
             bres = Parsers.xparse2(BigFloat, buf, startpos, len)
             if !Parsers.invalid(bres.code)
-                return NumberResult(bres.val), startpos + bres.tlen
+                return NumberResult(bres.val), startpos + Int(bres.tlen)
             end
         end
         if Parsers.invalid(res.code)
             error = InvalidNumber
             @goto invalid
         end
-        return NumberResult(res.val), startpos + res.tlen
+        return NumberResult(res.val), Int(startpos + res.tlen)
```

Comment:
Somewhat tangential: but what if this overflows to a negative position? I guess this means that on 32-bit systems we can't read files larger than 2 GB?

Reply:
Doing the `Int(startpos + res.tlen)` will throw an InexactError. Maybe ignore for now. Someone reading a massive file on a 32-bit system is asking for trouble anyway.